core/src/main/java/org/everit/json/schema/NumberSchemaValidatingVisitor.java

```diff
@@ -41,17 +42,17 @@ void visitMinimum(Number minimum) {
         if (minimum == null) {
             return;
         }
-        if (exclusiveMinimum && numberSubject <= minimum.doubleValue()) {
+        if (exclusiveMinimum && compare(numberSubject, minimum) <= 0) {
             owner.failure(subject + " is not greater than " + minimum, "exclusiveMinimum");
-        } else if (numberSubject < minimum.doubleValue()) {
+        } else if (compare(numberSubject, minimum) < 0) {
```

Comment:
Can you please extract the `compare(numberSubject, minimum)` value into a local variable?